client/api/InnerClient.cs

```diff
@@ -213,13 +238,25 @@ private void OnNotifyEvaluationChanged(string identifier)
         {
             EvaluationChanged?.Invoke(parent, identifier);
         }
+        
+        private void OnNotifyFlagsLoaded(IList<string> identifiers)
+        {
+            FlagsLoaded?.Invoke(parent, identifiers);
+        }
 
         public bool BoolVariation(string key, Target target, bool defaultValue)
         {
             try
             {
-                return evaluator.BoolVariation(key, target, defaultValue);
+                if (SdkInitialized) return evaluator.BoolVariation(key, target, defaultValue);
+
+                if (logger.IsEnabled(LogLevel.Warning))
```

Comment:
a) warning is v likely to be enabled
b) you are not allocating or calculating anything expensive to call LogWarning
so remove the guard and let it do that internally for practically identical cost even in the logging disabled scneario

Reply:
I think I'll opt to keep the guard, just to be on the safe side.  Evaluations are our critical path, so I would air on the side of caution here.  However, I do agree with your technical reasoning 👍

Reply:
there's nothing 'safe' about a redundant guard. its a micro perf optimization which makes the code harder to read and get right.
Log statements that get filtered are something that any real logger has well optimized - i.e. it needs to do an identical check first thing anyway for correctness
the guard is only for if you were going to calculate something, concatenate strings, or allocate
to call LogWarning you're placing 2 string refs into 2 registers
to call IsEnabled you're placing one enum value into 1 register
depending on whether you do one or the other, the next piece of work is
a) IsEnabled check log level > the value passed in the register
b) LogWarning check log level >= warning

There's a damn good chance that the second check will even be cheaper

Reply:
> the guard is only for if you were going to calculate something, concatenate strings, or allocate

Okay - point taken.  I agree it is harder to read and get right (evidenced by the wrong level being used in the guard earlier).  As I am not doing any expensive operations as you've called out, I'll remove the guards.  Thanks for the advice.